### PR TITLE
battop: init a version 0.2.4

### PR DESCRIPTION
**Pull request by hdhog**
###### Motivation for this change

Helpful tool for view power consumption on laptop

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/system/battop/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{ lib, stdenv, fetchFromGitHub, rustPlatform, installShellFiles }:
+rustPlatform.buildRustPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/system/battop/default.nix:2]:
```suggestion
{ lib, stdenv, fetchFromGitHub, rustPlatform, installShellFiles }:

rustPlatform.buildRustPackage rec {
```

File: pkgs/tools/system/battop/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    rev = "v${version}";
+    sha256 = "0p53jl3r2p1w9m2fvhzzrj8d9gwpzs22df5sbm7wwja4pxn7ay1w";
+  };
+
+  # https://github.com/svartalf/rust-battop/issues/11
+  cargoPatches = [ ./battery.patch ];
+
+  cargoSha256 = "0ipmnrn6lmf6rqzsqmaxzy9lblrxyrxzkji968356nxxmwzfbfvh";
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/tools/system/battop/default.nix:18]: Why did you disable tests?

File: pkgs/tools/system/battop/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  meta = with lib; {
+    description = ''
```
Comment (SuperSandro2000) [inline pkgs/tools/system/battop/default.nix:21]:
```suggestion
    description = "...";
    longDescription = ''
```
please set a short description.

File: pkgs/tools/system/battop/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      battop is an interactive viewer, similar to top,
+      htop and other *top utilities, but about
+      the batteries installed in your notebook.'';
+    homepage = "https://github.com/svartalf/rust-battop";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ hdhog ];
+    platforms = platforms.unix;
+    mainProgram = "battop";
```
Comment (SuperSandro2000) [inline pkgs/tools/system/battop/default.nix:29]:
```suggestion
```
both are defaults.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
The package is missing a top level entry.
please follow the contributing guide when creating our commits.

File: pkgs/tools/system/battop/default.nix (hunk 4 above)
Comment (hdhog) [reply pkgs/tools/system/battop/default.nix]: i removed it

File: pkgs/tools/system/battop/default.nix (hunk 2 above)
Comment (hdhog) [reply pkgs/tools/system/battop/default.nix]: it's my mistake, fixed

**Review by hdhog — COMMENTED** [review]
ok i fixed it

**Pull request merged**